src/index.ts

```diff
@@ -388,9 +393,11 @@ async function execute(
 }
 
 run()
-    .catch(core.setFailed)
-    .finally(() => {
+    .catch((err) => {
+        core.setFailed(err.message);
+    })
+    .finally(async () => {
         if (isImageFromDocker) {
-            removeDockerImage();
+            await removeDockerImage();
```

Comment:
looking at this now, this will result in an unhandled rejection if there is a problem

a better solution may be

```
run()
    .then(async () => {
        // await remove docker image here
    }
    catch((err) => {
// same as current
}
```

and leave out the `finally`

Reply:
this way an error thrown by removeDockerImage will be passed along to the catch block

Reply:
yeah